bindings/rust/s2n-tls/src/raw/connection/builder.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::raw::{
+    config::Config,
+    connection::{AsConnection, Connection},
+    enums::Mode,
+    error::Error,
+};
+
+/// A trait indicating that a structure can produce connections.
+pub trait Builder: Clone {
+    type Output: AsConnection;
```

Comment:
What's the purpose of supporting arbitrary outputs from the builder?

Reply:
For example, [connection pools](https://github.com/aws/s2n-tls/pull/3336). They don't return connections, because connections don't have the specific drop behavior they need. But they return something that can behave like a connection.